### PR TITLE
Check constancy when calling to external function

### LINKED ISSUE #1468: Bug: can't assert result of external calls to modifiable functions
### Version Information

* vyper Version (output of `vyper --version`): 0.1.0b10

### What's your issue about?
Upgrading Uniswap from vyper 0.1.0b4 to 0.1.0b10 and noticed my tests were failing on asserting external contract calls that return bools.

This fails (reverts):
```python
assert self.token.transferFrom(msg.sender, self, token_amount)
```

This succeeds:
```python
success: bool = self.token.transferFrom(msg.sender, self, token_amount)
assert success
```

The interface was defined as:
```
contract Token():
    def transferFrom(_from : address, _to : address, _value : uint256) -> bool: modifying

```

### PR DESCRIPTION
**Pull request by charles-cooper**
### What I did
Fix #1468 , #1377 

### How I did it
The compiler was not checking constancy of external calls inside of a constant context. However, it was correctly generating static calls, so calls to modifying functions would fail at runtime when the callee tried to modify state (per https://github.com/ethereum/EIPs/blob/d95612c8206b7e225bec5dc34c4b019c9531e44d/EIPS/eip-214.md). This adds a check when compiling external calls that they cannot be modifying within a constant context.

### Description for the changelog
Add constancy check for external calls

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: vyper/parser/external_call.py [hunk 1]
```diff
@@ -68,6 +69,15 @@ def external_contract_call(node,
         ['assert', ['extcodesize', contract_address]],
         ['assert', ['ne', 'address', contract_address]],
     ]
+    if context.is_constant and not sig.const:
+        raise ConstancyViolationException(
+            "May not call non-constant function '%s' within %s." % (
```
Comment (jacqueswww) [inline vyper/parser/external_call.py]: Add suggested use case here perhaps as well?
ie. "for checking returns of modifiable contract calls check ret: bool = self.Modify()
assert ret"

**Comment by charles-cooper** [comment]
Note to self: we decided at last meeting to add an assert_no_modify (or something) to the language and so this PR needs to be expanded to add that. Also I need to fix the tests

**Comment by charles-cooper** [comment]
@jacqueswww should be ready to go

**Comment by charles-cooper** [comment]
pinging @fubuloubu @davesque for review

**Comment by fubuloubu** [comment]
Need to add some docs on `assert_modifiable` but I don't see why this cannot be merged now.

**Pull request merged**